proposals/0304-structured-concurrency.md

```diff
@@ -155,30 +159,46 @@ Our approach follows the principles of *structured concurrency* described above.
 
 A *task group* defines a scope in which one can create new child tasks programmatically. As with all child tasks, the child tasks within the task group scope must complete when the scope exits, and will be implicitly cancelled first if the scope exits with a thrown error.
 
+Task groups are a low-level yet very powerful ab
+
 To illustrate task groups, let's start by showing how we can introduce some
 real concurrency to our `makeDinner` example:
 
 ```swift
-func makeDinner() async throws -> Meal {
+func makeDinner() async -> Meal {
   // Prepare some variables to receive results from our concurrent child tasks
   var veggies: [Vegetable]?
   var meat: Meat?
   var oven: Oven?
 
+  enum CookingStep { 
+    case veggies([Vegetable])
+    case meat(Meat)
+    case oven(Oven)
+  }
+  
   // Create a task group to scope the lifetime of our three child tasks
-  try await Task.withGroup(resultType: Void.self) { group in
-    await group.add {
-      veggies = try await chopVegetables()
+  await withTaskGroup(of: CookingStep.self) { group in
+    group.spawn {
+      try await .vegetables(chopVegetables())
     }
-    await group.add {
-      meat = await marinateMeat()
+    group.spawn {
```

Comment:
I discuss the removal of await here a bit below. 

The short version is:
- we never implemented this "suspend as a form of back-pressure" nor are we even confident it is the right thing to do, rather than some more holistic take on the concept.
- that await was tremendously confusing to reviewers -- "it is suspending until the operation completed?" was very frequently asked.